dashboard/app/controllers/pd/workshop_enrollment_controller.rb

```diff
@@ -45,6 +45,8 @@ def create
       render :full
     else
       @enrollment = ::Pd::Enrollment.new workshop: @workshop
+      @enrollment.build_school_info school_info_params
```

Comment:
You might be able to use `@enrollment.school_info = SchoolInfo.find_or_create_by(school_info_params)`